web/src/test/java/com/microsoft/applicationinsights/web/internal/httputils/HttpServerHandlerTest.java

```diff
@@ -0,0 +1,171 @@
+package com.microsoft.applicationinsights.web.internal.httputils;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.telemetry.RequestTelemetry;
+import com.microsoft.applicationinsights.web.internal.RequestTelemetryContext;
+import com.microsoft.applicationinsights.web.internal.ThreadContext;
+import com.microsoft.applicationinsights.web.internal.WebModulesContainer;
+import org.hamcrest.CoreMatchers;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.InOrder;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.net.MalformedURLException;
+import java.util.Date;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.StringContains.containsString;
+import static org.mockito.Mockito.*;
+
+@RunWith(JUnit4.class)
+public class HttpServerHandlerTest {
+
+    @Rule public final ExpectedException thrown = ExpectedException.none();
+    @Mock public HttpExtractor<HttpServletRequest, HttpServletResponse> extractor;
+    private TelemetryConfiguration telemetryConfiguration = TelemetryConfiguration.getActive();
+    @Spy public WebModulesContainer<HttpServletRequest,HttpServletResponse> webModulesContainer =
+        new WebModulesContainer<>(telemetryConfiguration);
+    @Spy public TelemetryClient telemetryClient;
+    @InjectMocks HttpServerHandler<HttpServletRequest, HttpServletResponse> httpServerHandler;
+    @Mock HttpServletRequest request;
+    @Mock HttpServletResponse response;
+    @Mock HttpServletRequest requestWithQueryString;
+    private String url = "http://www.abc.com/xyz/opq";
+    private String url1 = "http://30thh.loc:8480/app/test%3F/a%3F+b;jsessionid=S%3F+ID?p+1=c+d&p+2=e+f#a";
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        when(extractor.getUrl(request)).thenReturn(url);
+        when(extractor.getURI(request)).thenReturn("/xyz/opq");
```

Comment:
It is confusing when I read getUrl and getURI, do we have a better name, probably need to rethink the naming here.

Reply:
getUniformResourceLocator() and getUniformResourceIdentifier(). How does that sounds?

Reply:
renamed to above.

Reply:
@reyang https://javaee.github.io/javaee-spec/javadocs/javax/servlet/http/HttpServletRequest.html#getRequestURI-- 
As per this doc - getUri() returns part of URL and getURL() returns the full url that client used. Does this clarify your question.

Reply:
Got it, thank you.